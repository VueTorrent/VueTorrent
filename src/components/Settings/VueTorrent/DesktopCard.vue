<script setup lang="ts">

</script>

<template>
  VueTorrent > DesktopCard
</template>

<style scoped lang="scss">

</style>