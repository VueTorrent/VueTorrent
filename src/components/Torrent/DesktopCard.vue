<template>
  <v-layout row wrap class="ma-0 px-4 py-2 ml-0">
    <v-flex xs12>
      <div class="caption grey--text">
        {{ $t('torrent.torrentTitle') }}
      </div>
      <div class="truncate mr-4">
        {{ torrent.name }}
      </div>
    </v-flex>
    <component :is="item.name" v-for="item in properties" :key="item.name" :torrent="torrent" />
  </v-layout>
</template>
<script>
import * as Fields from './DashboardItems'
import { mapGetters } from 'vuex'

export default {
  name: 'DesktopCard',
  components: {
    ...Fields
  },
  props: {
    torrent: Object
  },
  computed: {
    ...mapGetters(['getWebuiSettings']),
    properties() {
      if (this.torrent.progress === 100) {
        return this.getWebuiSettings().doneTorrentProperties.filter(i => i.active)
      }

      return this.getWebuiSettings().busyTorrentProperties.filter(i => i.active)
    }
  }
}
</script>
