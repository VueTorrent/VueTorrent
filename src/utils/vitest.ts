import { mount } from '@vue/test-utils'
import { createVuetify } from 'vuetify'

export function setup(component: any, propsData?: any) {
  const vuetify = createVuetify() 

  return mount(component, {
    vuetify,
    propsData
  })
}