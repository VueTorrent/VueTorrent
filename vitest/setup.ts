export async function setup() {
  console.log('📝[vitest] globalSetup2.')
}

export async function teardown() {
  console.log('📝[vitest] globalTeardown')
}
