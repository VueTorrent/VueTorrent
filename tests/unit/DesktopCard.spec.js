import { describe, beforeEach, it, expect, vi } from 'vitest'
import { shallowMount } from '@vue/test-utils'

import DesktopCard from '@/components/Settings/Tabs/VueTorrent/VDesktopCard.vue'

let wrapper

describe('DesktopCard', () => {
  beforeEach(() => {
    wrapper = shallowMount(DesktopCard, {
      mocks: {
        $t: x => x,
        $store: {
          getters: { getAppVersion: vi.fn() },
          state: {
            webuiSettings: {
              lang: 'en',
              darkTheme: false,
              showFreeSpace: true,
              showSpeedGraph: true,
              showSessionStat: true,
              showAlltimeStat: true,
              showCurrentSpeed: true,
              showTrackerFilter: false,
              showSpeedInTitle: false,
              deleteWithFiles: false,
              title: 'Default',
              rightDrawer: false,
              paginationSize: 15,
              busyDesktopTorrentProperties: [
                { name: 'Size', active: true },
                { name: 'Progress', active: true },
                { name: 'Download', active: true },
                { name: 'Upload', active: true },
                { name: 'Downloaded', active: true },
                { name: 'Uploaded', active: true },
                { name: 'ETA', active: true },
                { name: 'Peers', active: true },
                { name: 'Seeds', active: true },
                { name: 'Status', active: true },
                { name: 'Ratio', active: true },
                { name: 'Category', active: true },
                { name: 'Tags', active: true },
                { name: 'AddedOn', active: true },
                { name: 'Availability', active: true }
              ],
              doneDesktopTorrentProperties: [
                { name: 'Size', active: true },
                { name: 'Progress', active: true },
                { name: 'Download', active: true },
                { name: 'Upload', active: true },
                { name: 'Downloaded', active: true },
                { name: 'Uploaded', active: true },
                { name: 'ETA', active: true },
                { name: 'Peers', active: true },
                { name: 'Seeds', active: true },
                { name: 'Status', active: true },
                { name: 'Ratio', active: true },
                { name: 'Category', active: true },
                { name: 'Tags', active: true },
                { name: 'AddedOn', active: true },
                { name: 'Availability', active: true }
              ]
            }
          }
        }
      }
    })
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('tests busyDesktopTorrentProperties', () => {
    const busyDesktopTorrentProperties = [
      { name: 'Size', label: 'Size', active: true },
      { name: 'Progress', label: 'Progress', active: true },
      { name: 'Download', label: 'Download Speed', active: true },
      { name: 'Upload', label: 'Upload Speed', active: true },
      { name: 'Downloaded', label: 'Downloaded', active: true },
      { name: 'Uploaded', label: 'Uploaded', active: true },
      { name: 'ETA', label: 'ETA', active: true },
      { name: 'Peers', label: 'Peers', active: true },
      { name: 'Seeds', label: 'Seeds', active: true },
      { name: 'Status', label: 'State', active: true },
      { name: 'Ratio', label: 'Ratio', active: true },
      { name: 'Category', label: 'Category', active: true },
      { name: 'Tags', label: 'Tags', active: true },
      { name: 'AddedOn', label: 'Added On', active: true },
      { name: 'Availability', label: 'Availability', active: true }
    ]
    expect(wrapper.vm.busyDesktopTorrentProperties).toEqual(busyDesktopTorrentProperties)
  })

  it('tests doneDesktopTorrentProperties', () => {
    const doneDesktopTorrentProperties = [
      { name: 'Size', label: 'Size', active: true },
      { name: 'Progress', label: 'Progress', active: true },
      { name: 'Download', label: 'Download Speed', active: true },
      { name: 'Upload', label: 'Upload Speed', active: true },
      { name: 'Downloaded', label: 'Downloaded', active: true },
      { name: 'Uploaded', label: 'Uploaded', active: true },
      { name: 'ETA', label: 'ETA', active: true },
      { name: 'Peers', label: 'Peers', active: true },
      { name: 'Seeds', label: 'Seeds', active: true },
      { name: 'Status', label: 'State', active: true },
      { name: 'Ratio', label: 'Ratio', active: true },
      { name: 'Category', label: 'Category', active: true },
      { name: 'Tags', label: 'Tags', active: true },
      { name: 'AddedOn', label: 'Added On', active: true },
      { name: 'Availability', label: 'Availability', active: true }
    ]
    expect(wrapper.vm.doneDesktopTorrentProperties).toEqual(doneDesktopTorrentProperties)
  })
})
