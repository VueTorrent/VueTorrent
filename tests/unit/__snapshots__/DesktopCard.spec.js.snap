// Vitest Snapshot v1

exports[`DesktopCard > render correctly 1`] = `
"<v-card-stub loaderheight=\\"4\\" tag=\\"div\\" flat=\\"true\\">
  <v-row-stub tag=\\"div\\" dense=\\"true\\">
    <v-col-stub cols=\\"12\\" md=\\"6\\" tag=\\"div\\">
      <v-subheader-stub> modals.settings.vueTorrent.desktopCard.busyTorrentTip </v-subheader-stub>
      <v-row-stub tag=\\"div\\" dense=\\"true\\">
        <v-list-stub tag=\\"div\\" flat=\\"true\\" class=\\"ma-2 pa-0\\">
          <draggable-stub list=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" clone=\\"[Function]\\" element=\\"div\\" tag=\\"tbody\\">
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
          </draggable-stub>
        </v-list-stub>
      </v-row-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" md=\\"6\\" tag=\\"div\\">
      <v-subheader-stub> modals.settings.vueTorrent.desktopCard.completedTorrentTip </v-subheader-stub>
      <v-row-stub tag=\\"div\\" dense=\\"true\\">
        <v-list-stub tag=\\"div\\" flat=\\"true\\" class=\\"ma-2 pa-0\\">
          <draggable-stub list=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" clone=\\"[Function]\\" element=\\"div\\" tag=\\"tbody\\">
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
            <vdashboarditem-stub isdraggable=\\"true\\" property=\\"[object Object]\\"></vdashboarditem-stub>
          </draggable-stub>
        </v-list-stub>
      </v-row-stub>
    </v-col-stub>
  </v-row-stub>
</v-card-stub>"
`;
